src/wx/widgets/sdljoy.cpp

```diff
@@ -38,420 +36,455 @@ static int16_t axisval(int16_t x)
+                SDL_GameControllerGetButton(
+                    game_controller_,
+                    static_cast<SDL_GameControllerButton>(but));
+
+            if (previous_value != current_value)
+                ProcessEvent(SDL_CONTROLLERBUTTONUP, but, current_value);
+        }
 
-                    systemScreenMessage(wxString::Format(_("Disconnected game controller %d"), index));
-                }
+        for (uint8_t axis = 0; axis < SDL_CONTROLLER_AXIS_MAX; axis++) {
+            uint16_t previous_value = axis_[axis];
+            uint16_t current_value =
+                AxisValueToDirection(
+                    SDL_GameControllerGetAxis(
+                        game_controller_,
+                        static_cast<SDL_GameControllerAxis>(axis)));
 
-                got_event = true;
+            if (previous_value != current_value)
+                ProcessEvent(SDL_CONTROLLERAXISMOTION, axis, current_value);
+        }
+    } else {
+        for (uint8_t but = 0; but < SDL_JoystickNumButtons(joystick_); but++) {
+            uint16_t previous_value = buttons_[but];
+            uint16_t current_value = SDL_JoystickGetButton(joystick_, but);
 
-                break;
-            }
+            if (previous_value != current_value)
+                ProcessEvent(SDL_JOYBUTTONUP, but, current_value);
+        }
 
-            // Joystck events for non-GameControllers.
+        for (uint8_t axis = 0; axis < SDL_JoystickNumAxes(joystick_); axis++) {
+            uint16_t previous_value = axis_[axis];
+            uint16_t current_value =
+                AxisValueToDirection(SDL_JoystickGetButton(joystick_, axis));
 
-            case SDL_JOYBUTTONDOWN:
-            case SDL_JOYBUTTONUP:
-            {
-                auto joy = e.jbutton.which;
+            if (previous_value != current_value)
+                ProcessEvent(SDL_JOYAXISMOTION, axis, current_value);
+        }
 
-                if (contains(instance_map, joy)) {
-                    auto& state   = *(instance_map[joy]);
+        for (uint8_t hat = 0; hat < SDL_JoystickNumHats(joystick_); hat++) {
+            uint16_t previous_value = hats_[hat];
+            uint16_t current_value = SDL_JoystickGetHat(joystick_, hat);
 
-                    if (SDL_IsGameController(state.index))
-                        break;
+            if (previous_value != current_value)
+                ProcessEvent(SDL_JOYHATMOTION, hat, current_value);
+        }
+    }
+}
 
-                    auto but      = e.jbutton.button;
-                    auto val      = e.jbutton.state;
-                    auto prev_val = state.button[but];
+void wxSDLJoyState::SetRumble(bool activate_rumble) {
+    if (!IsGameController())
+        return;
 
-                    if (val != prev_val) {
-                        CreateAndSendEvent(state.index, WXSDLJOY_BUTTON, but, val, prev_val);
+    if (activate_rumble)
+        SDL_GameControllerRumble(game_controller_, 0xFFFF, 0xFFFF, 300);
```

Comment:
You need to check SDL version here, this is why the CI is failing.

Reply:
Yes, the SDL Check was for the API, silly me :)